src/libraries/Microsoft.Extensions.Configuration.Binder/tests/SourceGenerationTests/Baselines/TestBindCallGen.generated.txt

```diff
@@ -0,0 +1,122 @@
+// <auto-generated/>
+#nullable enable
+
+using System.Linq;
+
+internal static class GeneratedConfigurationBinder
+{
+    public static void Bind(this global::Microsoft.Extensions.Configuration.IConfiguration configuration, Program.MyClass obj) => BindCore(configuration, ref obj);
+
+    private static void BindCore(global::Microsoft.Extensions.Configuration.IConfiguration configuration, ref Program.MyClass obj)
+    {
+        if (obj is null)
+        {
+            throw new global::System.ArgumentNullException(nameof(obj));
+        }
+
+        if (configuration["MyString"] is string stringValue0)
+        {
+            obj.MyString = stringValue0;
+        }
+
+        if (configuration["MyInt"] is string stringValue1)
+        {
+            obj.MyInt = int.Parse(stringValue1);
+        }
+
+        System.Collections.Generic.List<int> temp2 = obj.MyList;
```

Comment:
We are inconsistent on prefixing types with `global::`.

It would be better if we followed the same pattern in Regex and the RequestDelegateGenerator to put all the generated logic in a namesapce, with `using` statements inside the namespace declaration, and a `file class` implementation. That allows us to generate code that looks like how a user will write it.

See https://github.com/dotnet/aspnetcore/pull/46180#discussion_r1082824958 and the 2nd bullet of https://github.com/dotnet/runtime/pull/66432.

Reply:
Resolved in https://github.com/dotnet/runtime/pull/83614.